Upgrade to go 1.10

derekwbrown opened the pull request:
Try with dummy file

Add pkg_config_path to source tests

Pick specific test docker image for testing with go1.10.2

Change both tests to go110

derekwbrown committed:
Add printf debugging

Try with dummy file

Add pkg_config_path to source tests

Pick specific test docker image for testing with go1.10.2

Change both tests to go110

derekwbrown committed:
Make logging changes

olivielpeau commented on pkg/util/xc/dummy.go:
```diff
@@ -0,0 +1,10 @@
+
+func doNothing() int64 {
+	return 0
+}
```
probably a temp file you can remove now :)

derekwbrown replied:
No.  go vet complains that pkg/util/xc doesn't contain any files otherwise.

olivielpeau reviewed (commented):
Just one nit

The fixes on the logger are also part of this PR: https://github.com/DataDog/datadog-agent/pull/1758

Looks like the gitlab pipeline is failing on the dogstatsd size test because the binary is now 18MB large (instead of ~13MB with go 1.9), probably interesting to look into it quickly and sync with Kenafeh on this.

derekwbrown committed:
Bump max binary size to 20M to compensate for go 1.10.
File card to look at stripping binary, at least on docker

derekwbrown committed:
Take cluster agent out on windows

olivielpeau reviewed (commented):
A few more comments, once addressed the PR LGTM 👍

hush-hush committed:
Fix 'vet' command on windows (#1791)

Since LDFLAGS are filter by a whitelst for cgo
(https://github.com/golang/go/wiki/InvalidFlag) the vet command failed
on windows with go 1.10. We now use the same flags for 'vet' than for
building (or else pkg/collector/py would raise an "invalid flag in #cgo
CFLAGS" error).

derekwbrown committed:
add release note

derekwbrown committed:
Merge branch 'db/fixgo110' of github.com:DataDog/datadog-agent into db/fixgo110

derekwbrown committed:
review feedback

derekwbrown committed:
Revert explicit setting of PKG_CONFIG_PATH for test builds

derekwbrown committed:
Revert change of specific gitlab runner to latest
